plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryPageSink.java

```diff
@@ -104,7 +105,8 @@ public CompletableFuture<?> appendPage(Page page)
 
     private void insertWithCommitted(JSONArray batch)
     {
-        try (JsonStreamWriter writer = JsonStreamWriter.newBuilder(writeStream.getName(), writeStream.getTableSchema(), client).build()) {
+        WriteStream stream = writeStream.updateAndGet(current -> current != null ? current : client.createWriteStream(createWriteStreamRequest));
```

Comment:
it's not support readable for me in this form. Can we write in some other way around?

Reply:
Can we extract this to a separate method: 
```
    private WriteStream getOrCreateWriteStream(WriteStream current)
    {
        if (current == null) {
            return client.createWriteStream(createWriteStreamRequest));
        }
        return current;
    }
```